swap/src/protocol/alice/event_loop.rs

```diff
@@ -161,8 +161,38 @@ where
                                     continue;
                                 }
                             }
+                            // TODO: This should be cleaned up.
```

Comment:
Note: 
In my PR [1] that introduced a dedicated spot_price behaviour for Alice I still construct `State0` outside. We could move that inside the spot_price behaviour and just return `State0` directly. I will see to tackle this once we have one of them in and go for rebase :)

[1] https://github.com/comit-network/xmr-btc-swap/pull/461/files#diff-ac04e254d84094077194f78ec27926a36a2ceb76e65b40d719e7d9ab5b547254

Reply:
I gave this a bit more thought. Network behaviours are not async, so I don't think we can actually move that piece of code into . 
Sidenote: We use an async-await network behaviour for the `execution_setup` and we could move the construction of `State0` in there, but I am not sure that is a good design. The `execution_setup` behaviour currently purely focuses on the message exchange and expects `State0` from the outside. I am not sure it should be concerned about the construction of that state.

Thus, at the moment I don't see a good way to move interaction with the wallet out of the eventloop. 
Given the current design I think it will have to stay similar to what it is now, so you could actually remove the ToDo.

Reply:
Interesting. 
Yea, the fee-estimation should not be moved outside the loop because we should only compute it when needed. I'll remove the todo and wait for your PR to introduce the new behavior. Maybe it becomes obvious what can be done here.